### PR TITLE
exchange.parseAccount changed to typeToAccount

### PR DESCRIPTION
**Pull request by samgermain**
`parseAccount` is a different method

```
% gateio transfer USDT 1 spot ADA/USDT                                                                                 % gateio transfer USDT 1 spot CROSS                                                                                      % gateio transfer USDT 1 CROSS spot
2022-08-11T22:11:26.287Z                                                                                               2022-08-11T22:13:22.879Z                                                                                                 2022-08-11T22:13:34.079Z
Node.js: v18.4.0                                                                                                       Node.js: v18.4.0                                                                                                         Node.js: v18.4.0
CCXT v1.92.14                                                                                                          CCXT v1.92.14                                                                                                            CCXT v1.92.14
(node:33915) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time      (node:34003) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time        (node:34013) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)                                                (Use `node --trace-warnings ...` to show where the warning was created)                                                  (Use `node --trace-warnings ...` to show where the warning was created)
gateio.transfer (USDT, 1, spot, ADA/USDT)                                                                              gateio.transfer (USDT, 1, spot, CROSS)                                                                                   gateio.transfer (USDT, 1, CROSS, spot)
2022-08-11T22:11:29.104Z iteration 0 passed in 231 ms                                                                  2022-08-11T22:13:24.993Z iteration 0 passed in 235 ms                                                                    2022-08-11T22:13:35.957Z iteration 0 passed in 224 ms
                                                                                                                                                                                                                                                
{                                                                                                                      {                                                                                                                        {
  id: undefined,                                                                                                         id: undefined,                                                                                                           id: undefined,
  timestamp: 1660255889104,                                                                                              timestamp: 1660256004993,                                                                                                timestamp: 1660256015957,
  datetime: '2022-08-11T22:11:29.104Z',                                                                                  datetime: '2022-08-11T22:13:24.993Z',                                                                                    datetime: '2022-08-11T22:13:35.957Z',
  currency: 'USDT',                                                                                                      currency: 'USDT',                                                                                                        currency: 'USDT',
  amount: 1,                                                                                                             amount: 1,                                                                                                               amount: 1,
  fromAccount: 'spot',                                                                                                   fromAccount: 'spot',                                                                                                     fromAccount: 'CROSS',
  toAccount: 'ADA/USDT',                                                                                                 toAccount: 'CROSS',                                                                                                      toAccount: 'spot',
  status: undefined,                                                                                                     status: undefined,                                                                                                       status: undefined,
  info: { tx_id: '103551043' }                                                                                           info: { tx_id: '1660256005024256' }                                                                                      info: { tx_id: '1660256015986967' }
}                                                                                                                      }                                                                                                                        }
2022-08-11T22:11:29.104Z iteration 1 passed in 231 ms                                                                  2022-08-11T22:13:24.993Z iteration 1 passed in 235 ms                                                                    2022-08-11T22:13:35.957Z iteration 1 passed in 224 ms
```

### CODE REVIEW

File: js/base/Exchange.js [hunk 1]
```diff
@@ -2682,7 +2682,7 @@ module.exports = class Exchange {
         return undefined;
     }
 
-    parseAccount (account) {
```
Comment (carlosmiei) [inline js/base/Exchange.js:2685]: What's the reasoning behind this renaming?

File: js/okx.js [hunk 2]
```diff
@@ -1976,14 +1976,14 @@ module.exports = class okx extends Exchange {
         /**
          * @method
          * @name okx#createOrder
-         * @description create a trade order
+         * @description create a trade order, *reduceOnly must be specified for swap/future to reduce a position*
```
Comment (carlosmiei) [inline js/okx.js:1979]: These changes are completely unrelated, please revert them

File: js/base/Exchange.js (hunk 1 above)
Comment (samgermain) [reply js/base/Exchange.js:2685]: `parseAccount` is already a method that serves a different purpose, it's written on a number of exchanges

Comment (carlosmiei) [reply js/base/Exchange.js:2685]: @samgermain I see but I don't find `typeToAccount` a very clear name, @kroitor are you ok with this name?

File: js/okx.js (hunk 2 above)
Comment (carlosmiei) [reply js/okx.js:1979]: @samgermain also don't forget to revert this

Comment (samgermain) [reply js/okx.js]: In the future, if it's a simple commit like this that you don't think should be removed, could you just merge it anyway. When important commits like this that other commits are built on don't get merged, it really slows down progress

Comment (carlosmiei) [reply js/okx.js]: @samgermain I'm not sure I fully understand your point but I'm insisting on getting all unrelated changes removed because if something goes wrong and we need to revert it we can pick up the merge commit and revert the entire PR really quickly, if you have here unrelated unchanged those would be reverted as well unnecessarily

File: js/gate.js [hunk 3]
```diff
@@ -3870,8 +3870,8 @@ module.exports = class gate extends Exchange {
          */
         await this.loadMarkets ();
         const currency = this.currency (code);
-        const fromId = this.parseAccount (fromAccount);
-        const toId = this.parseAccount (toAccount);
+        const fromId = this.typeToAccount (fromAccount);
```
Comment (carlosmiei) [inline js/gate.js:3873]: @samgermain you renamed in the base but not its invocations

**Review by carlosmiei — CHANGES_REQUESTED** [review]
Missing fixing the rename

**Comment by carlosmiei** [comment]
LGTM ✅ ✅ 

```
 p  gateio transfer USDT 1 spot ADA/USDT  
Python v3.9.13
CCXT v1.92.30
gateio.transfer(USDT,1,spot,ADA/USDT)
{'amount': 1.0,
 'currency': 'USDT',
 'datetime': '2022-08-18T07:17:28.902Z',
 'fromAccount': 'spot',
 'id': None,
 'info': {'tx_id': '104193322'},
 'status': None,
 'timestamp': 1660807048902,
 'toAccount': 'ADA/USDT'}
 ```
 
```
 p  gateio transfer USDT 1 spot futures
Python v3.9.13
CCXT v1.92.30
gateio.transfer(USDT,1,spot,futures)
{'amount': 1.0,
 'currency': 'USDT',
 'datetime': '2022-08-18T07:18:44.716Z',
 'fromAccount': 'spot',
 'id': None,
 'info': {'tx_id': '1660807124'},
 'status': None,
 'timestamp': 1660807124716,
 'toAccount': 'futures'}
 ```

```
p  gateio transfer USDT 1 futures spot
Python v3.9.13
CCXT v1.92.30
gateio.transfer(USDT,1,futures,spot)
{'amount': 1.0,
 'currency': 'USDT',
 'datetime': '2022-08-18T07:19:26.432Z',
 'fromAccount': 'futures',
 'id': None,
 'info': {'tx_id': '1660807166'},
 'status': None,
 'timestamp': 1660807166432,
 'toAccount': 'spot'}

```

```
p kucoin transfer USDT 1 main contract
Python v3.9.13
CCXT v1.92.30
kucoin.transfer(USDT,1,main,contract)
{'amount': '1',
 'currency': 'USDT',
 'datetime': None,
 'fromAccount': 'main',
 'id': '62fde855d56cfa0001f8fcee',
 'info': {'orderId': '62fde855d56cfa0001f8fcee'},
 'status': None,
 'timestamp': None,
 'toAccount': 'contract'}
 ```

**Review by carlosmiei — APPROVED** [review]
